packages/shared/lib/core/router/subrouters/backup-router.ts

```diff
@@ -15,44 +15,60 @@ export const backupRoute = writable<BackupRoute>(null)
 
 export class BackupRouter extends Subrouter<BackupRoute> {
     constructor() {
-        super(BackupRoute.Init, backupRoute)
+        if (get(mobile)) {
+            requestMnemonic()
+            super(BackupRoute.RecoveryPhrase, backupRoute)
+        } else {
+            super(BackupRoute.Init, backupRoute)
+        }
     }
 
     async next(event: FireflyEvent): Promise<void> {
         let nextRoute: BackupRoute
 
         const currentRoute = get(this.routeStore)
-        switch (currentRoute) {
-            case BackupRoute.Init:
-                await requestMnemonic()
-                nextRoute = BackupRoute.RecoveryPhrase
-                break
-            case BackupRoute.RecoveryPhrase:
-                nextRoute = BackupRoute.Verify
-                break
-
-            case BackupRoute.Verify:
-                nextRoute = BackupRoute.Backup
-                break
-
-            case BackupRoute.Backup:
-                if (event?.skip) {
+        if (get(mobile)) {
+            switch (currentRoute) {
```

Comment:
That looks a bit messy. Perhaps we can put the switch statements in separat functions? Don't know, maybe handleMobileRouting and handleDesktopRouting